products/iap/api.yaml

```diff
@@ -0,0 +1,75 @@
+# Copyright 2019 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: Iap
+display_name: Identity-Aware Proxy
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://iap.googleapis.com/v1/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Cloud Identity-Aware Proxy
+    url: https://console.cloud.google.com/apis/library/iap.googleapis.com/
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Web'
+    base_url: 'projects/{{project}}/iap_web'
+    self_link: 'projects/{{project}}/iap_web'
+    exclude_resource: true
+    description: |
+      Only used to generate IAM resources
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: Dummy property.
+        required: true
+  - !ruby/object:Api::Resource
+    name: 'WebTypeCompute'
+    base_url: 'projects/{{project}}/iap_web/compute'
+    self_link: 'projects/{{project}}/iap_web/compute'
+    exclude_resource: true
+    description: |
+      Only used to generate IAM resources
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: Dummy property.
+        required: true
+  - !ruby/object:Api::Resource
+    name: 'WebTypeAppEngine'
+    base_url: 'projects/{{project}}/iap_web/appengine-{{appEngineId}}'
+    self_link: 'projects/{{project}}/iap_web/appengine-{{appEngineId}}'
+    exclude_resource: true
+    description: |
+      Only used to generate IAM resources
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'appEngineId'
+        description: Id of the App Engine application.
+        required: true
+  - !ruby/object:Api::Resource
+    name: 'WebBackendService'
+    base_url: 'projects/{{project}}/iap_web/compute/services/{{backendServiceName}}'
+    self_link: 'projects/{{project}}/iap_web/compute/services/{{backendServiceName}}'
+    exclude_resource: true
+    description: |
+      Only used to generate IAM resources
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'backendServiceName'
+        description: Name or ID of a backend service.
```

Comment:
eh let's say name or self link so we don't give people ideas about using the `.id` property